### PR TITLE
002 view job spec

### PR DESCRIPTION
**Pull request by 40175096**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
   "main": "index.js",
   "scripts": {
     "start": "npx tsc && cp -R views dist && node dist/app.js",
-    "test": "ts-mocha -p ./tsconfig.json ./test/**/**/*.ts",
```
Comment (eoinmcbennett) [inline package.json:8]: This line needs to stay the same i think

**Review by eoinmcbennett — CHANGES_REQUESTED** [review]

**Review by eoinmcbennett — APPROVED** [review]

File: test/unit/service/jobServiceTest.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    });
+
+    // throw exception when no data returned from API
+    it('should throw exception when empty data returned from axios', async () => {
+      let error;
+      // mock axios
+      const mock = new MockAdapter(axios);
+      const id: number = 1;
+      // return no data when call made to API
+      mock.onGet(`${API_URL}/api/job-specification/${id}`).reply();
```
Comment (shaunganley) [inline test/unit/service/jobServiceTest.ts:122]: I'm assuming job does not exist would return a 404 from API?

File: test/ui/job-specification.test.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        await button.click();
+        
+        const jobTable = await driver.wait(until.elementLocated(By.id('jobs')), 30000);
+        expect(await jobTable.isDisplayed()).to.be.true;
+      });
+    });
+   
+    describe('"First View in Sharepoint" button', function() {
+      it('should redirect to the appropriate page when clicked', async function() {
+        const buttons = await driver.findElements(By.id('viewSharePointButton'));
```
Comment (shaunganley) [inline test/ui/job-specification.test.ts]: should have a unique ID per button

File: test/ui/job-specification.test.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    this.timeout(100000);
+   
+    let driver: WebDriver;
+   
+    before(async function() {
+      const options = new chrome.Options();
+      options.addArguments('--headless');
+      options.addArguments('--window-size=1920,1080');
+      driver = await new Builder().forBrowser('chrome').setChromeOptions(options).build();
+      await driver.get('http://localhost:3000/job-specification/:id');
```
Comment (shaunganley) [inline test/ui/job-specification.test.ts]: pass the base URL in as an env var

**Review by shaunganley — APPROVED** [review]

File: test/unit/service/jobServiceTest.ts (hunk 2 above)
Comment (40175096) [reply test/unit/service/jobServiceTest.ts:122]: No, it just returns 200 with no jobs and the front end checks if no jobs are returned

**Pull request merged**